### PR TITLE
Fixed MAT-146 + Technical Debt cleanup

### PR DESCRIPTION
**Pull request by cday-semanticbits**
Fixed MAT-146. What was going on was there were two spring beans and getBean was being called which threw an exception preventing cql exports. 

To fix this i went ahead and did some technical debt cleanup. I removed all the xml configured beans and did it all with annotations. After doing this I also had to go in and fix the places where the bean names were referenced.

### CODE REVIEW

File: src/main/java/mat/client/export/measure/ManageMeasureExportView.java [hunk 1]
```diff
@@ -127,6 +128,17 @@ public void setExportOptionsBasedOnVersion(String releaseVersion, boolean isComp
 		vp.add(compositeMeasurePackageRadio);
 		resetRadioButtonValues(isCompositeMeasure);
 	}
+
+	private boolean isV5OrGreater(String version) {
+		boolean result = false;
+		try {
+			result = Double.parseDouble(version.substring(1,version.length())) >= 5.0;
+		} catch (RuntimeException re) {
+		}
```
Comment (serhii-ilin) [inline src/main/java/mat/client/export/measure/ManageMeasureExportView.java]: Codacy will complain on a catch-and-swallow block. It should have a comment or a logging statement

File: src/main/java/mat/server/Application.java [hunk 2]
```diff
@@ -56,14 +57,14 @@
...
 @Configuration
-@ComponentScan({"mat.model", "mat.dao", "mat.dao.impl", "mat.model.clause", "mat.server", "mat.hibernate"})
 @PropertySource("classpath:MAT.properties")
 @EnableTransactionManagement
 @EnableConfigurationProperties
 @EnableWebSecurity
 @EnableCaching
 @EnableScheduling
 @EnableJpaRepositories
+@Service
```
Comment (serhii-ilin) [inline src/main/java/mat/server/Application.java]: why do you need a service annotation on Application class? It's probably not needed

Comment (cday-semanticbits) [reply src/main/java/mat/server/Application.java]: It's a war remember, not a spring boot application. That just ensures it gets scanned.

File: src/main/java/mat/client/export/measure/ManageMeasureExportView.java (hunk 1 above)
Comment (cday-semanticbits) [reply src/main/java/mat/client/export/measure/ManageMeasureExportView.java]: ok will add a comment.

File: src/main/java/mat/client/export/measure/ManageMeasureExportView.java [hunk 3]
```diff
@@ -127,6 +128,17 @@ public void setExportOptionsBasedOnVersion(String releaseVersion, boolean isComp
 		vp.add(compositeMeasurePackageRadio);
 		resetRadioButtonValues(isCompositeMeasure);
 	}
+
+	private boolean isV5OrGreater(String version) {
+		boolean result = false;
+		try {
+			result = Double.parseDouble(version.substring(1,version.length())) >= 5.0;
+		} catch (RuntimeException re) {
```
Comment (jkotanchik-SB) [inline src/main/java/mat/client/export/measure/ManageMeasureExportView.java]: `RuntimeException` seems overly broad. Could it be some combination of other exceptions? If we get up to a runtime exception, there's probably something else we should be doing.

File: src/main/java/mat/server/clause/QDSAttributesServiceImpl.java [hunk 4]
```diff
@@ -29,11 +29,13 @@
 import mat.model.clause.QDSAttributes;
 import mat.server.SpringRemoteServiceServlet;
 import mat.server.util.ResourceLoader;
+import org.springframework.stereotype.Service;
 
 /**
  * The Class QDSAttributesServiceImpl.
  */
 @SuppressWarnings("serial")
+@Service
```
Comment (serhii-ilin) [inline src/main/java/mat/server/clause/QDSAttributesServiceImpl.java]: It's a servlet. It should not have a Service annotation (if it's not injected somewhere else).

File: src/main/java/mat/client/export/measure/ManageMeasureExportView.java (hunk 3 above)
Comment (cday-semanticbits) [reply src/main/java/mat/client/export/measure/ManageMeasureExportView.java]: It handles all the cases I need NullPointer,NumberFormat, etc. Why create multiple blocks and increase the code size?

File: src/main/java/mat/server/clause/QDSAttributesServiceImpl.java (hunk 4 above)
Comment (cday-semanticbits) [reply src/main/java/mat/server/clause/QDSAttributesServiceImpl.java]: Wat its not a servlet? its a spring bean.

File: src/main/java/mat/server/Application.java (hunk 2 above)
Comment (serhii-ilin) [reply src/main/java/mat/server/Application.java]: I think it has "@Configuration" annotation which should be scanned by default.

**Comment by serhii-ilin** [comment]
overall looks good to me.  a couple of minor remarks and complains from codacy

File: src/main/java/mat/client/export/measure/ManageMeasureExportView.java (hunk 3 above)
Comment (serhii-ilin) [reply src/main/java/mat/client/export/measure/ManageMeasureExportView.java]: It's java 11.  catch(NullPointer|IllegalArgumentException e) { // ... }
Indeed. Most of code scanning toold will complain about a borad exception :)

File: src/main/java/mat/server/Application.java (hunk 2 above)
Comment (jkotanchik-SB) [reply src/main/java/mat/server/Application.java]: From [here](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Configuration.html):

> @Configuration is meta-annotated with @Component, therefore @Configuration classes are candidates for component scanning

File: src/main/java/mat/server/service/impl/MeasurePackageServiceImpl.java [hunk 5]
```diff
@@ -122,7 +125,8 @@
     @Autowired
     private CompositeMeasurePackageValidator compositeMeasurePackageValidator;
 
-    private String currentReleaseVersion;
+    @Value("${mat.measure.current.release.version}")
+    private String currentReleaseVersion = "v6.0";
```
Comment (jkotanchik-SB) [inline src/main/java/mat/server/service/impl/MeasurePackageServiceImpl.java]: What happens when you assign a value and use the value annotation?

File: src/main/java/mat/server/service/impl/SimpleEMeasureServiceImpl.java [hunk 6]
```diff
@@ -246,7 +246,7 @@ private void findAndAddDTO(final NodeList nodeList, final List<QualityDataSetDTO
 	 * @return Measure. *
 	 */
 	private mat.model.clause.Measure getMeasureName(final String measureId) {
-		MeasurePackageService measureService = (MeasurePackageService) context.getBean("measurePackageService");
+		MeasurePackageService measureService = (MeasurePackageService) context.getBean(MeasurePackageService.class);
```
Comment (jkotanchik-SB) [inline src/main/java/mat/server/service/impl/SimpleEMeasureServiceImpl.java]: Can remove the type cast.

File: src/main/java/mat/server/service/impl/MeasurePackageServiceImpl.java (hunk 5 above)
Comment (cday-semanticbits) [reply src/main/java/mat/server/service/impl/MeasurePackageServiceImpl.java]: it overrides what was there from the property. need to remove the string.

File: src/main/java/mat/server/service/impl/SimpleEMeasureServiceImpl.java (hunk 6 above)
Comment (cday-semanticbits) [reply src/main/java/mat/server/service/impl/SimpleEMeasureServiceImpl.java]: removing

File: src/main/java/mat/server/clause/QDSAttributesServiceImpl.java (hunk 4 above)
Comment (cday-semanticbits) [reply src/main/java/mat/server/clause/QDSAttributesServiceImpl.java]: See what you mean now, my bad. It was a bean before but prob someone new to spring. Going to try removing it and see if it works.

File: src/main/java/mat/client/export/measure/ManageMeasureExportView.java (hunk 3 above)
Comment (cday-semanticbits) [reply src/main/java/mat/client/export/measure/ManageMeasureExportView.java]: nice, switched over to java11 style.

File: src/main/java/mat/server/clause/QDSAttributesServiceImpl.java (hunk 4 above)
Comment (cday-semanticbits) [reply src/main/java/mat/server/clause/QDSAttributesServiceImpl.java]: It worked no need for it to be a bean.

File: src/main/java/mat/client/export/measure/ManageMeasureExportView.java (hunk 3 above)
Comment (serhii-ilin) [reply src/main/java/mat/client/export/measure/ManageMeasureExportView.java]: sorry for misleading you.
it's a Java7 feature
https://docs.oracle.com/javase/7/docs/technotes/guides/language/catch-multiple.html

**Pull request merged**